### PR TITLE
Enka automation

### PR DESCRIPTION
**Pull request by ChinPTeo**

### CODE REVIEW

**Comment by ChinPTeo** [comment]
@srliao can't find config.yml files for artifacts, are they manual or generated auto?

**Comment by srliao** [comment]
> @srliao can't find config.yml files for artifacts, are they manual or generated auto?

Ah shoot my bad. I remember now. I left out artifact pipeline at the time because I been meaning to ask you what data exactly will you be needing for artifacts. Was it only the names? I think we were only using the names to come up with the key right? If so, then is there any other data we actually need to extract out of dm?

Also there is one more blocking issues re traveler (see https://github.com/genshinsim/gcsim/pull/1419#issuecomment-1626988194). Basically need to split out traveler into individual folders for each lumine and aether. I'll try and get to this mid next week

File: pkg/core/keys/sets.go [hunk 1]
```diff
@@ -80,6 +80,9 @@ var setNames = []string{
 	"wandererstroupe",
 }
 
+//export setNames
+var SetNames = setNames
```
Comment (k0l11) [inline pkg/core/keys/sets.go:84]: Why not just capitalize it during initialization?

File: protos/model/data.proto [hunk 2]
```diff
@@ -70,7 +70,10 @@ message ArtifactDataMap {
   map<string, ArtifactData> data = 1 [ json_name = "data" ];
 }
 
-message ArtifactData { int32 id = 1 [ json_name = "id" ]; }
+message ArtifactData { 
+  int32 id = 1 [ json_name = "id" ]; 
+  string text_map_id = 2 [ json_name = "text_map_id"];
+}
```
Comment (k0l11) [inline protos/model/data.proto:76]: Doesn't look like this matches the structure of how artifact data is currently generated.

File: ui/packages/ui/src/Pages/Simulator/Components/Enka/EnkaToGOOD.ts [hunk 3]
```diff
@@ -1,91 +1,186 @@
...
+let artifactMapByTextMapId: { [key in string]: string } = {};
+for (const [k, v] of Object.entries(ArtifactDataGen.data)) {
+  artifactMapByTextMapId[v.text_map_id] = k;
+}
+
+const travelerSkillIdToEleMap: {
+  skill_id: number;
+  sub_id: number;
+}[] = Object.values(CharDataGen.data)
+  .filter((e) => e.key.includes("aether"))
```
Comment (k0l11) [inline ui/packages/ui/src/Pages/Simulator/Components/Enka/EnkaToGOOD.ts:74]: Is checking only for aether really enough?

**Review by k0l11 — APPROVED** [review]

**Pull request merged**